### PR TITLE
add in range reason enum. fixes #8

### LINKED ISSUE #8: Expand on Range Reason Enum
https://github.com/NODE-Collective/specification/blob/1e5595660f329a6259f4f7e27f43cff2e39d4e4e/SPECIFICATION.md?plain=1#L212

### PR DESCRIPTION
**Pull request by FredericoAndrade**

### CODE REVIEW

File: SPECIFICATION.md [hunk 1]
```diff
@@ -211,7 +211,21 @@ TODO
 
 ### range_reason
 
-TODO
+| Value | Description |
+|--|--|
+| Empty |  |
+| Income | Applicant income impacts the amount of the incentive |
+| InstallerQualifications | Installer qualifications impact the amount of the incentive |
+| PurchaseLocation | Specific vendor impacts the amount of the incentive |
```
Comment (oyamauchi) [inline SPECIFICATION.md]: I thought this was about where you buy the thing, not which specific vendor you buy from? (E.g. must buy your EV in ABC County)

Well, I think I have seen incentives where you have to buy from an authorized distributor or something, but that feels like a separate category from a restriction on the physical location of the vendor. Maybe we need another enum value?

File: SPECIFICATION.md [hunk 2]
```diff
@@ -211,7 +211,21 @@ TODO
...
+| InstallationLocation | Location the technology is installed impacts the amount of the incentive|
+| Membership | Applicant's membership or association impacts the amount of the incentive |
+| NotApplicable |  |
+| Other |  |
+| ReplacementType | The type of replacement impacts the amount of the incentive |
```
Comment (oyamauchi) [inline SPECIFICATION.md]: Is this more like what type of equipment is _being replaced_?

File: SPECIFICATION.md [hunk 3]
```diff
@@ -211,7 +211,21 @@ TODO
...
+| Unknown |  |
+| UpgradeMeasureCost | The cost of the technoogy impacts the amount of the incentive |
+| UpgradeMeasureSpecifications | The specifications of the technoogy impacts the amount of the incentive |
```
Comment (oyamauchi) [inline SPECIFICATION.md]: Not sure we need both words "upgrade" and "measure"; I think just `MeasureSpecifications` would be enough?

Also `technoogy`

File: SPECIFICATION.md [hunk 4]
```diff
@@ -211,7 +211,21 @@ TODO
...
+| Income | Applicant income impacts the amount of the incentive |
+| InstallerQualifications | Installer qualifications impact the amount of the incentive |
+| PurchaseLocation | Specific vendor impacts the amount of the incentive |
+| InstallationLocation | Location the technology is installed impacts the amount of the incentive|
+| Membership | Applicant's membership or association impacts the amount of the incentive |
+| NotApplicable |  |
+| Other |  |
+| ReplacementType | The type of replacement impacts the amount of the incentive |
+| Unknown |  |
+| UpgradeMeasureCost | The cost of the technoogy impacts the amount of the incentive |
```
Comment (oyamauchi) [inline SPECIFICATION.md]: Why is this one necessary? Wouldn't this be expressed as a `Normalized` amount record?

File: SPECIFICATION.md [hunk 5]
```diff
@@ -211,7 +211,21 @@ TODO
 
 ### range_reason
 
-TODO
+| Value | Description |
+|--|--|
+| Empty |  |
```
Comment (oyamauchi) [inline SPECIFICATION.md]: I don't think we need this; other enums that appear in optional fields don't have an "empty" entry. I think we don't need `NotApplicable` either.

**Review by oyamauchi — DISMISSED** [review]
I kinda wonder if we shouldn't have both `Other` and `Unknown`? I'm OK with `Other`, but I kinda think if there's a factor for an incentive amount varying that's literally _unknown_, maybe it shouldn't be included at all? Or maybe we could just code that as `Other`?

File: SPECIFICATION.md (hunk 1 above)
Comment (kswb) [reply SPECIFICATION.md]: Authorized distributors are somewhat common for EVs and there are sometimes for local incentive bonuses for in-city purchases, validating purchase location but for a region not a vendor.

File: SPECIFICATION.md (hunk 2 above)
Comment (kswb) [reply SPECIFICATION.md]: Flagging that equipment type replacement matters as well as fuel type (gas -> electrified vs. electric -> electric)

**Review by mpusto — APPROVED** [review]
Looks good, thanks for cleaning this up!

**Pull request merged**